webapp/app/forms/flows/multistep_flow.py

```diff
@@ -131,11 +131,10 @@ def url_for_step(self, step_name, _has_link_overview=None, **values):
                        **values)
 
     def _get_session_data(self, ttl: Optional[int] = None):
-        serialized_session = session.get('form_data', b"")
-        session_data = deserialize_session_data(serialized_session, ttl)
+        form_data = get_session_data('form_data', ttl)
```

Comment:
Could we use get_session_data to set the default data, too? It provides the functionality and we would not duplicate the code here.
```suggestion
        form_data = get_session_data('form_data', ttl, self.default_data()[1]) if self.default_data() else get_session_data('form_data', ttl)
```

Or delete the whole method and directly use get_session_data in line 85, which I would prefer.
```python
 stored_data = get_session_data('form_data', ttl, self.default_data()[1]) if self.default_data() else get_session_data('form_data', ttl)
```

Reply:
I agree with Jördis here. Not because you can put everything in 1 line you should do it. Readability should also be taken into consideration!

Reply:
No, we do not need it here, because session should user bounded at this point. We have to check it in the new redis client implementation.

Reply:
The part about how debug data and session data is combined is imo duplicated as we have it in `_get_session_data` and in `get_session_data`:
`form_data = self.default_data()[1] | form_data  # updates form_data only with non_existent values`

And if we try to only use session_data.py everywhere, I would like to get rid of similar implementations and have a single source of truth how debug data is put together with the session data.

> I think, if you put all code in one line, it will be less readable and more complex because of an additional else what is not necessary and you should avoid if possible.

Yes, I see that. The problem is that self.default_data gives us a tuple and not a dict or None. 


What about something like that:
```python
default_data = self.default_data()[1] if self.default_data() else None
stored_data = get_session_data('form_data', ttl, default_data)
```

Reply:
get_session_data should be the global user scoped wrapper fct to handle data and the other one is a private fct and is only scoped for this multistep class and should handle data inside this class (handling form_data only). So  they are handling/describing the same concept, but in different scopes.  In fact the same concept should be named the same, so there is no confusing.

Reply:
> Not because you can put everything in 1 line you should do it. Readability should also be taken into consideration!

I agree with that consideration.

Nevertheless, I did not want to make the point that we could write it in one line. I rather wanted to to only handle the update of the session data only in session_data.py as I think how the combination of the default_data and session_data happens should not be a concern of the multistep_flow. This was only the best solution I could come up with to get rid of the _get_session_data.

Basically, I think of something like the following scenario:
- We don't want to add deleted keys to the session data if they are in the default data
- If we change that, we have to change it in _get_session_data and in get_session_data

> get_session_data should be the global user scoped wrapper fct to handle data and the other one is a private fct and is only scoped for this multistep class and should handle data inside this class (handling form_data only). So they are handling/describing the same concept, but in different scopes. In fact the same concept should be named the same, so there is no confusing.

Okay, I might not see why we would need to have a private fct for the multistep that basically does the same. Imo either the private function would prepare the arguments to call the global fct or would have a different behavior. What would you have in mind that we would need that function?

However, I think it is a rather minor detail and I would be happy to a) either talking about it or b) you merging that PR and if we need to refactor it in the future we will deal with it when the time comes

Reply:
> 

default_data should not be handled by the session_data. But you are right with the default_data in multistep. Maybe we could refactor it and move it to a separate service.